### PR TITLE
Wcs metric

### PR DESCRIPTION
**Pull request by fergusL**

### CODE REVIEW

File: src/huntsman/drp/quality/metrics/rawexp.py [hunk 1]
```diff
@@ -1,9 +1,58 @@
...
+RAW_METRICS = ("has_wcs", "clipped_stats", "flipped_asymmetry")
 
 
-def clipped_stats(data, file_info, **kwargs):
+def get_wcs(filename, timeout=60, downsample=4, radius=5, *args):
+    """Function to call get_solve_field on a file and verify
+    if a WCS solution could be found.
+
+    Args:
+         filename (str): The filename.
```
Comment (danjampro) [inline src/huntsman/drp/quality/metrics/rawexp.py]: Inconsistent indentation

File: src/huntsman/drp/quality/metrics/rawexp.py [hunk 2]
```diff
@@ -1,9 +1,58 @@
...
+        pass
+
+    # if file is not a science exposure, skip
+    if parsed_hdr['dataType'] != "science":
+        return {"has_wcs": has_wcs}
+
+    # Create list of args to pass to solve_field
+    solve_kwargs = {'--cpulimit': str(timeout),
+                    '--downsample': downsample,
+                    '--ra': ra,
```
Comment (danjampro) [inline src/huntsman/drp/quality/metrics/rawexp.py]: If the header does not have `RA-MNT/DEC-MNT` then `ra`/`dec` will not be set and there will be an error.

File: src/huntsman/drp/quality/metrics/rawexp.py [hunk 3]
```diff
@@ -1,9 +1,58 @@
...
+                    '--dec': dec,
+                    '--radius': radius}
+    # now solve for wcs
+    try:
+        get_solve_field(filename, *args, **solve_kwargs)
+    except Exception:
+        pass
```
Comment (danjampro) [inline src/huntsman/drp/quality/metrics/rawexp.py:50]: Should we log an error message here?

File: src/huntsman/drp/quality/metrics/rawexp.py [hunk 4]
```diff
@@ -1,9 +1,58 @@
...
+
+    # finally check if the header now contians a wcs solution
+    wcs = WCS(read_fits_header(filename))
+    if wcs.has_celestial:
```
Comment (danjampro) [inline src/huntsman/drp/quality/metrics/rawexp.py]: Can condense this to: 
```
return {"has_wcs": wcs.has_celestial}
```

File: src/huntsman/drp/quality/screening.py [hunk 5]
```diff
@@ -35,6 +34,9 @@ def __init__(self, exposure_table=None, sleep_interval=None, status_interval=60,
         """
         super().__init__(*args, **kwargs)
 
+        # work around so that tests can run without running the has_wcs metric
+        self.RAW_METRICS = RAW_METRICS
```
Comment (danjampro) [inline src/huntsman/drp/quality/screening.py]: I don't think this needs to be capitalised. We should probably also make a copy of the original if we plan to edit it within the class:

`self._raw_metrics = deepcopy(RAW_METRICS)`

File: src/huntsman/drp/tests/test_screener.py [hunk 6]
```diff
@@ -17,33 +16,38 @@ def test_screener_ingest(tempdir_and_exposure_table_with_uningested_files, confi
...
 
-    m = Screener(exposure_table=exposure_table, sleep_interval=0.1, status_interval=0.1,
-                 monitored_directory=tempdir)
+    screener = Screener(exposure_table=exposure_table, sleep_interval=0.1, status_interval=0.1,
+                        monitored_directory=tempdir)
+    # don't want to run astrometry tasks as tests running in drp-lsst container
+    # not the drp container
+    testable_raw_metrics = list(screener.RAW_METRICS)
+    testable_raw_metrics.remove('has_wcs')
+    screener.RAW_METRICS = tuple(testable_raw_metrics)
```
Comment (danjampro) [inline src/huntsman/drp/tests/test_screener.py]: Could do this in one line if you wanted to:
`screener._raw_metrics = [_ for _ in screener._raw_metrics if _ != "has_wcs"]`

**Review by danjampro — APPROVED** [review]
Looks good! See comment about potential for error if ra/dec are not set.

File: src/huntsman/drp/quality/metrics/rawexp.py (hunk 3 above)
Comment (fergusL) [reply src/huntsman/drp/quality/metrics/rawexp.py:50]: I think before you were opposed to passing logger objects in to the metric functions, can't remember why. I personally don't mind either way

**Pull request merged**